subprojects/testing-jvm/src/main/java/org/gradle/api/internal/tasks/testing/junit/ParameterizedRunnerCategoryFilter.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks.testing.junit;
+
+import org.junit.runner.Description;
+
+import java.util.Set;
+
+/**
+ * {@link org.junit.runners.Parameterized} runner wraps the original {@link Description}s with some top-level wrapper descriptions (which are named like "[0]", "[1]").
+ * These wrapper description should not participate in filtering.
+ */
+class ParameterizedRunnerCategoryFilter extends CategoryFilter {
```

Comment:
❌  I'm not too happy with adding more and more special cases for different runners. According to the JUnit [issue tracker](https://github.com/junit-team/junit4/issues/751) this was fixed in 4.13. If that works, we should just ask users to upgrade instead of putting more workarounds in our code base.